### ISSUE TITLE
Android blocker inserter entrance animation occasionally jumps to middle of screen

### ISSUE THREAD

**Issue by dcalhoun** [WordPress/gutenberg#32012]
## Description
When opening the block inserter on Android, the animation often flashes the blocker inserter UI onto the screen before subsequently hiding it and animating _upwards_ into place. Occasionally, the block inserter will even render detached from the bottom of the screen, leaving it floating in the middle of the screen before animating _downward_ into place.

The issue is fairly sporadic and difficult to reproduce consistently, at least the detached block inserter occurrence. It has been reproduced in the WPAndroid 17.3 beta build. It was not reproducible in the WPAndroid 17.2 Play Store build. 

## Step-by-step reproduction instructions
1. Launch the Gutenberg block editor in the WPAndroid app. 
1. Tap the plus icon button to open the block inserter. 
1. Close the block inserter. 
1. Repeat the previous steps multiple times over. 

## Expected behaviour
The block inserter always animates into place sliding _upwards_.

## Actual behaviour
The block inserter often flashes on the screen before animating. Occasionally animates _downwards_ to the bottom of the screen. 

## Screenshots or screen recording (optional)

<details><summary>Screenshot of detached block inserter</summary>

<img width="652" alt="android-block-inserter-detached" src="https://user-images.githubusercontent.com/438664/118863648-c5f13a00-b8a4-11eb-8f40-7860de6881d2.png">

</details>

<details><summary>17.3 beta recording with jumping animation (occurs on 4th and final attempt)</summary>

https://user-images.githubusercontent.com/438664/118863355-698e1a80-b8a4-11eb-92d4-2b072b2732bc.mp4

</details>

<details><summary>17.2 recording without jumping animation</summary>

https://user-images.githubusercontent.com/438664/118863435-7f034480-b8a4-11eb-9749-280a6d056fc4.mp4

</details>

## WordPress information
- WordPress version: 5.6.0
- Gutenberg version: 10.5.4
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

## Device information
- Device: Samsung Galaxy S20
- Operating system: Android 11
- WordPress app version: 17.3

**Comment by dcalhoun** [comment]
From reviewing [the 1.53 release changes](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3506), it appears a few changes (#30291, #31655, #28495) were made to the block inserter. At this time, I am unsure if any of them are directly related. The next best step might be to individually revert each change to identify where the issue may have originated.

**Comment by fluiddot** [comment]
That's interesting, as you pointed, most likely this was introduced on those PRs. I'll try to spare some time to investigate this further, thanks for opening the issue 🙇 !

**Comment by jordesign** [comment]
@dcalhoun Just looping through some older issues and wanted to check if this bug is still relevant?

**Comment by dcalhoun** [comment]
@jordesign yes, this bug is still present. It is less noticeable and occurs in a slightly different manner now that the inserter is full screen height on Android only (https://github.com/WordPress/gutenberg/pull/34129), but it the issue remains. 

We can keep this open or close it and consider this one of many performance/animation issues to resolve in https://github.com/WordPress/gutenberg/issues/37559. Either approach is valid in my opinion. I updated https://github.com/WordPress/gutenberg/issues/37559 to reference this issue for posterity.

**Comment by tyxla** [comment]
This is no longer relevant after the React Native implementation was removed in #78747. Closing.
